### PR TITLE
config: Add default paths for project config file

### LINKED ISSUE #425: chaire-lib: Add a default path for the project config file
The PROJECT_CONFIG environment variable needs to point to the configuration file. We should have a default path where to look for that file if it is not set, for example first CWD, then maybe $HOME/.config/transition/ then /etc/transition.

### PR DESCRIPTION
**Pull request by tahini**
Fixes #425

First, we check if the PROJECT_CONFIG environment variable contains the
path to the config file. Otherwise, it is expected to be in the
$homedir/.config/transition/config.js or /etc/transition/config.js.

The path in the .env now has to be absolute. ${rootDir} works only in
the current repository, but chaire-lib is meant as a library, so rootDir
will not work for those.

Add a start:demo script in the package.json which sets the
PROJECT_CONFIG environment variable to the example config.

### CODE REVIEW

**Review by kaligrafy — APPROVED** [review]

**Comment by tahini** [comment]
I added the fix of #459 to this PR

TLDR:

1. The PROJECT_CONFIG environment variable can be set, either in .env or some other way, but if not set, it will look in $HOME/.config/transition/config.js or /etc/transition/config.js
2. The TRANSITION_DOTENV environment variable can be set, defaults to <this repo's root>/.env. Thus it does not need to be set in this repo, but in transition-legacy yes (tested and it works now)

The doc has not been updated yet, I'll re-work it in the next PRs, with complete startup documentation.

@greenscientist Can you take a look with deployment and infra in mind? (knowing #452 and #97 still need fixing)

**Comment by greenscientist** [comment]
One quick comment. PROJECT_CONFIG path being absolute is fine, but one alternative way is to treat a string starting with a "/" as absolute and one without it as relateive to the CWD. But we can always replicate the later by setting the string to $(CWD)/something. 

For TRANSITION_DOTENV, " defaults to <this repo's root>/.env. " what does happen when you run the code from outside a git repo, like a TGZ that packaged everything? Or a docker image? (Probably need an answer to this before commiting)

**Comment by tahini** [comment]
I was expecting that any repo NOT running from here would require those environment variables, but true, I could use node's provided envirionment variables for CWD. I'll test that it's the correct one in every case and update if so

**Comment by tahini** [comment]
So... CWD and PWD happen to be packages/chaire-lib-backend because the `yarn workspace chaire-lib-backend` apparently chagned the working directory. But INIT_CWD is the directory from which the original command was run, ie the root of the repos that interest us. So the default relative paths are now set to INIT_CWD. It works both with git submodule in transition-legacy and here.

**Review by greenscientist — APPROVED** [review]
ok, sounds good
